### PR TITLE
Update runtime and dependencies to the latest versions

### PR DESCRIPTION
**Pull request by trivikr**
This PR does the following:
* Removes v2 since it's in [maintenance mode](https://aws.amazon.com/blogs/developer/announcing-end-of-support-for-aws-sdk-for-javascript-v2/), and provides reference to `v2` branch.
* Recommends using Node.js 22.x, as it's the current LTS version.
* Switches from yarn to npm with devEngines field.
* Update dependencies to latest versions.
* Switch from prettier to biome.
* Update screenshots from Jaeger.
* Updates deprecated objects from OpenTelemetry.

### CODE REVIEW

File: src/tracing.js [hunk 1]
```diff
@@ -1,29 +1,27 @@
 import { AwsInstrumentation } from "@opentelemetry/instrumentation-aws-sdk";
 import { Resource } from "@opentelemetry/resources";
-import { NodeTracerProvider } from "@opentelemetry/sdk-trace-node";
 import {
-  ConsoleSpanExporter,
-  SimpleSpanProcessor,
+	ConsoleSpanExporter,
```
Comment (kuhe) [inline src/tracing.js]: this is a tab space? 😨

File: .editorconfig [hunk 2]
```diff
@@ -1,10 +0,0 @@
-root = true
```
Comment (trivikr) [inline .editorconfig]: Re-add this and check if biome reads it.

File: src/tracing.js (hunk 1 above)
Comment (trivikr) [reply src/tracing.js]: Looks like biome uses tabs by default.

I re-introduced `.editorconfig` and instructed biome to use that instead
[`a75287d` (#10)](https://github.com/trivikr/aws-sdk-js-otel-test/pull/10/commits/a75287df9b730e3f65ac9d2d4ddb4abf04c0ed3a)

File: .editorconfig (hunk 2 above)
Comment (trivikr) [reply .editorconfig]: I re-introduced `.editorconfig` and instructed biome to use that instead
[`a75287d` (#10)](https://github.com/trivikr/aws-sdk-js-otel-test/pull/10/commits/a75287df9b730e3f65ac9d2d4ddb4abf04c0ed3a)

**Pull request merged**